client/client.go

```diff
@@ -156,6 +157,13 @@ func (c *Client) read(greeted <-chan struct{}) error {
 		if err == io.EOF || c.State() == imap.LogoutState {
 			return nil
 		} else if err != nil {
+			if opErr, ok := err.(*net.OpError); ok {
+				if syscallErr, ok := opErr.Err.(*os.SyscallError); ok {
+					if syscallErr.Err == syscall.ECONNRESET {
```

Comment:
How did you end up hitting this error? (Just wondering whether this is the right fix)

Reply:
It sometimes happens when running the client tests.